lib/kubernetes-deploy/kubernetes_resource.rb

```diff
@@ -54,19 +41,28 @@ def timeout
       self.class.timeout
     end
 
-    def initialize(name:, namespace:, context:, file:, logger:)
+    def initialize(namespace:, context:, template:, logger:)
       # subclasses must also set these if they define their own initializer
-      @name = name
+      @name = template.fetch("metadata", {})["name"]
       @namespace = namespace
       @context = context
-      @file = file
       @logger = logger
+      @template = template
     end
 
     def id
       "#{type}/#{name}"
     end
 
+    def file
```

Comment:
This code is copy-pasted from what we're using today. It is primarily used for [passing its path to kubectl](https://github.com/Shopify/kubernetes-deploy/pull/113/files#diff-ae1efd633a69ca93e2ad9f56702555ebR230) when we do the dry run and when we actually deploy the resource. However, we may in fact read it [here](https://github.com/Shopify/kubernetes-deploy/pull/113/files#diff-ae1efd633a69ca93e2ad9f56702555ebR189) for debugging purposes. `file.read` would raise that error, but `File.read(file.path)` works fine, and that's what we're doing.